all_test.go

```diff
@@ -65,7 +66,25 @@ func TestShorty(t *testing.T) {
 		if d.URL != urls[i] {
 			t.Fatalf("expected %s, got %s", urls[i], d.URL)
 		}
+		if d.ID != ids[i] {
+			t.Fatalf("expected %s, got %s", urls[i], d.URL)
+		}
 		return nil
 	})
 
 }
+
+func TestExpiry(t *testing.T) {
+	var s Shorty
+	s.s = NewMemStore() // have to do it this way so the auto clean up doesn't start
+	id, err := s.GenerateTimedID("http://google.com", time.Second)
+	if err != nil {
+		t.Fatal(err)
+	}
+	time.Sleep(time.Second)
+	go s.cleanup() // this leaks but it's fine for the test
```

Comment:
Just so I understand why, is there a reason we do this in a goroutine instead of just letting it happen synchronously?